test/parallel/test-http2-compat-expect-continue.js

```diff
@@ -0,0 +1,54 @@
+// Flags: --expose-http2
+'use strict';
+
+const common = require('../common');
+if (!common.hasCrypto)
+  common.skip('missing crypto');
+const assert = require('assert');
+const http2 = require('http2');
+
+const testResBody = 'other stuff!\n';
+
+const server = http2.createServer();
+
+server.on('request', common.mustCall((req, res) => {
+  console.error('Server sent full response');
+
+  res.end(testResBody);
+}));
+
+server.listen(0);
+
+server.on('listening', common.mustCall(() => {
+  let body = '';
+
+  const port = server.address().port;
+  const client = http2.connect(`http://localhost:${port}`);
+  const req = client.request({
+    ':method': 'POST',
+    ':path': '/world',
+    expect: '100-continue'
+  });
+  console.error('Client sent request');
+
+  req.on('continue', common.mustCall(() => {
+    console.error('Client received 100-continue');
+  }));
+
+  req.on('response', common.mustCall((headers) => {
+    console.error('Client received response headers');
+
+    assert.strictEqual(headers[':status'], 200);
+  }));
+
+  req.on('data', common.mustCall((chunk) => { body += chunk; }));
```

Comment:
you will need `setEncoding('utf8')` to concatenate the body like this. `chunk` should be a `Buffer` here.

Reply:
Great point, totally missed it! Thanks.